source/plugin/internals/access-transformers.rst

```diff
@@ -0,0 +1,67 @@
+===================
+Access Transformers
+===================
+
+Since some parts of the Minecraft code were not designed to be used from the outside, you may find yourself in a
+situation in which you need to access a field or method that is not public. While you would normally use reflection to
+access the field or method, MCP will make this more difficult since you have two different names - the MCP names in the
+development environment and the Searge names in production.
+
+For example to access the method ``tick()`` using reflection you would need to use ``tick`` in the development
+environment, but ``func_71217_p`` in production. The re-obfuscation step only handles direct references to methods
+and fields, not the string parameter passed to the reflection call.
+
+As a solution, ForgeGradle supports using access transformers (or AT) that automatically make the specified
+methods/fields public so you can reference them directly (without reflection). If configured in the JAR manifest of the
+plugin, SpongeVanilla and Forge will also apply them in production.
+
+Setup
+-----
+ForgeGradle will automatically scan for access transformer files with the file name suffix ``_at.cfg`` in your resource
+folders. To be able to use the access transformer at runtime, you need to add them to a ``META-INF`` folder in your
+resource directory, for example ``META-INF/myplugin_at.cfg``.
+
+An access transformer line is defined by 3 parts, each separated by a space:
```

Comment:
Correct me if I'm wrong, but I believe there's actually 2 different types:

- Field/Method transformers, with the transformer, class and field/function
- Class transformers, with just the transformer and class (at least that's what I understood looking at the Forge AT code)

Might be worth mentioning, Minecraft probably has a few private classes that people might want to transform.